Parse/SMTLIB2.hpp

```diff
@@ -35,7 +35,16 @@ using namespace Shell;
 
 class SMTLIB2 {
 public:
-  SMTLIB2(const Options& opts);
+  /**
+   * @brief Construct a new SMTLIB2 parser
+   *
+   * @param formulaBuffer is FIFO to which newly parsed Formulas will be added (via pushBack);
+   *
+   *  if left unspeficied, and empty fifo is created and used instead.
+   *  (use this default behaviour if you do not want to collect formulas
+   *  from multiple parser calls)
+   */
+  SMTLIB2(UnitList::FIFO formulaBuffer = UnitList::FIFO());
```

Comment:
You want to write to the end and read from the beginning. (The TPTP parser was already doing this, so I didn't want to break this. I made the SMTLIB2 parser do the same - it probably should have been done that way from the outset.)

Well, you could minimally be given a `UnitList*&`, so that you can grow the list at the end yourself, but the FIFO is just mildly more than that and a bit better encapsulated. 

(Somehow I did stick with UnitList, but a Stack could work too.)

Reply:
OK - lightly consider a stack, the allocation profile would be more pleasing (and maybe usability is better as well). But I don't care enough to change it!